src/Task.ts

```diff
@@ -684,6 +683,99 @@ export class Task {
         return linkText;
     }
 
+    /**
+     * Compare two lists of Task objects, and report whether their
+     * tasks are identical in the same order.
+     *
+     * This can be useful for optimising code if it is guaranteed that
+     * there are no possible differences in the tasks in a file
+     * after an edit, for example.
+     *
+     * If any field is different in any task, it will return false.
+     *
+     * @param oldTasks
+     * @param newTasks
+     */
+    static tasksListsIdentical(oldTasks: Task[], newTasks: Task[]): boolean {
+        if (oldTasks.length !== newTasks.length) {
+            return false;
+        }
+        return oldTasks.every((oldTask, index) =>
+            oldTask.identicalTo(newTasks[index]),
+        );
+    }
+
+    /**
+     * Compare all the fields in another Task, to detect any differences from this one.
+     *
+     * If any field is different in any way, it will return false.
+     *
+     * This is used in some optimisations, to avoid work if an edit to file
+     * does not change any tasks, so it is vital that its definition
+     * of identical is very strict.
+     *
+     * @param other
+     */
+    public identicalTo(other: Task) {
+        // Based on ideas from koala. AquaCat and javalent in Discord:
+        // https://discord.com/channels/686053708261228577/840286264964022302/996735200388186182
+        // and later.
+        let args: Array<keyof Task> = [
```

Comment:
I think I should add a check for if the two tasks are the same object, as then there is no point comparing all the fields.

Reply:
Decided not to do that at this point. The performance is already a significant improvement, and I just want to get 1.9.0 released.